integration_test.go

```diff
@@ -0,0 +1,475 @@
+			"mode":              reflect.ValueOf(subAlgo).String(),
+			"orb_n_features":    "1250",
+			"orb_scale_factor":  "1.2",
+			"orb_n_levels":      "8",
+			"orb_n_ini_th_fast": "20",
+			"orb_n_min_th_fast": "7",
+			"debug":             "true",
+		},
+		DataDirectory: name,
+		// Even though we don't use the maps saved in this run, indicate in the config that
+		// we want to save maps because the same yaml config gets used for the next run.
+		MapRateSec:          &mapRate,
+		DeleteProcessedData: &deleteProcessedData,
+		UseLiveData:         &useLiveData,
+	}
+
+	// Release camera image(s) for service validation
+	releaseImages(t, subAlgo)
+	// Create slam service using a real orbslam binary
+	svc, err := createSLAMService(t, attrCfg, logger, true, true, viamorbslam3.DefaultExecutableName)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Release camera image(s), since orbslam looks for the second most recent image(s)
+	releaseImages(t, subAlgo)
+	// Check if orbslam hangs and needs to be shut down
+	orbslamHangs := false
+
+	// Wait for orbslam to finish processing images
+	logReader := svc.(testhelper.Service).GetSLAMProcessBufferedLogReader()
+	for i := 0; i < getNumOrbslamImages(subAlgo)-2; i++ {
+		startTimeSentImage := time.Now()
+		t.Logf("Find log line for image %v", i)
+		releaseImages(t, subAlgo)
+		for {
+			line, err := logReader.ReadString('\n')
+			test.That(t, err, test.ShouldBeNil)
+			if strings.Contains(line, "Passed image to SLAM") {
+				prevNumFiles = slamTesthelper.CheckDeleteProcessedData(t, slam.Mode(subAlgo), name, prevNumFiles, deleteProcessedData, useLiveData)
+				break
+			}
+			test.That(t, strings.Contains(line, "Fail to track local map!"), test.ShouldBeFalse)
+			if time.Since(startTimeSentImage) > time.Duration(dataInsertionMaxTimeoutMin)*time.Minute {
+				orbslamHangs = true
+				t.Log("orbslam hangs: exiting the data loop")
+				break
+			}
+		}
+		if orbslamHangs {
+			break
+		}
+	}
+
+	testOrbslamPosition(t, svc, reflect.ValueOf(subAlgo).String(), "mapping", attrCfg.Sensors[0])
+	testOrbslamMap(t, svc)
+
+	// Close out slam service
+	err = utils.TryClose(context.Background(), svc)
+	if !orbslamHangs {
+		test.That(t, err, test.ShouldBeNil)
+	} else if err != nil {
+		t.Skip("Skipping test because orbslam hangs and failed to shut down")
+	}
+
+	// Don't clear out the directory, since we will re-use the config and data for the next run
+	closeOutSLAMService(t, "")
+
+	// Added sleep to ensure orbslam stops
+	time.Sleep(time.Millisecond * orbSleepMs)
+
+	// test orbslam directory, should have 2 configs
+	testOrbslamDir(t, name, expectedMapsOnline, 2)
+
+	// Delete the last image (or image pair) in the data directory, so that offline mode runs on
+	// the same data as online mode. (Online mode will not read the last image (or image pair),
+	// since it always processes the second-most-recent image (or image pair), in case the
+	// most-recent image (or image pair) is currently being written.
```

Comment:
```suggestion
	// Delete the last image (or image pair) in the data directory so that offline mode runs on
	// the same data as online mode. Note: Online mode will not read the last image (or image pair),
	// since it always processes the second-most-recent image (or image pair), in case the
	//most recent image (or image pair) is still being written.
```